internal/controller/organization/organization_controller_test.go

```diff
@@ -341,38 +344,38 @@ var _ = Describe("Test Organization reconciliation", Ordered, func() {
-				for _, orgClient := range oAuthClients.Items {
+				filteredOAuthClients := slices.DeleteFunc(oAuthClients.Items, func(c dexapi.OAuth2Client) bool {
+					return c.ID != oidcOrg.Name && c.ID != defaultOrg.Name
+				})
+				Expect(filteredOAuthClients).To(HaveLen(2), "there should be two dex oauth clients")
+				for _, orgClient := range filteredOAuthClients {
 					switch orgClient.ID {
-					case oidcOrgName:
-						Expect(orgClient.ID).To(Equal(oidcOrgName), "the oauth client ID should be equal to organization name")
+					case oidcOrg.Name:
+						Expect(orgClient.ID).To(Equal(oidcOrg.Name), "the oauth client ID should be equal to organization name")
 						Expect(orgClient.RedirectURIs).To(HaveLen(5), "the oauth client redirect URIs should have the default 3 elements + 2 additionalRedirects")
 						Expect(orgClient.RedirectURIs).To(ContainElements("https://example.com/app", "http://localhost:33768/auth/callback"), "the oauth client redirect URIs should be equal to organization redirect URIs")
-					case greenhouseOrgName:
-						Expect(orgClient.ID).To(Equal(greenhouseOrgName), "the oauth client ID should be equal to organization name")
+					case defaultOrg.Name:
+						Expect(orgClient.ID).To(Equal(defaultOrg.Name), "the oauth client ID should be equal to organization name")
 						Expect(orgClient.RedirectURIs).To(ContainElements("https://test-oidc-org.dashboard."), "the greenhouse client should contain the org's dashboard redirect uri")
-						Expect(orgClient.RedirectURIs).To(HaveLen(5), "the oauth client redirect URIs should have 4 elements (default 3 + 1 org + 1 additional)")
+						Expect(len(orgClient.RedirectURIs)).To(BeNumerically(">=", 5), "the oauth client redirect URIs should have 4 elements (default 3 + 1 org + 1 additional)")
 					default:
 						Fail("unexpected oauth client ID")
 					}
 				}
 				By("deleting the organizations")
-				test.EventuallyDeleted(test.Ctx, test.K8sClient, &greenhousev1alpha1.Organization{ObjectMeta: metav1.ObjectMeta{Name: oidcOrgName}})
-				test.EventuallyDeleted(test.Ctx, test.K8sClient, &greenhousev1alpha1.Organization{ObjectMeta: metav1.ObjectMeta{Name: greenhouseOrgName}})
-
+				test.EventuallyDeleted(test.Ctx, test.K8sClient, &greenhousev1alpha1.Organization{ObjectMeta: metav1.ObjectMeta{Name: oidcOrg.Name}})
 				By("checking if the dex resources are deleted")
-				err = setup.List(test.Ctx, connectors)
-				Expect(err).ToNot(HaveOccurred(), "there should be no error listing dex connectors")
-				Expect(connectors.Items).To(BeEmpty(), "there should be no dex connector resources")
 				err = setup.List(test.Ctx, oAuthClients)
 				Expect(err).ToNot(HaveOccurred(), "there should be no error listing dex oauth clients")
-				Expect(oAuthClients.Items).To(BeEmpty(), "there should be no dex oauth clients resources")
+				for _, orgClient := range oAuthClients.Items {
+					Expect(orgClient.ID).ToNot(Equal(oidcOrg.Name), "the oauth client ID should be equal to organization name")
+				}
```

Comment:
NIT: Could do this with gomega. The expectation message seems off. I think it should be sth. `the oauth client ID of the deleted org should not exist`?
```suggestion
		                Expect(oAuthClients.Items).ToNot(ContainElement(
					MatchFields(IgnoreExtras, Fields{"ID": Equal(oidcOrg.Name)}),
				), "the oauth client list should not contain the deleted organization")
```